packit_service/worker/build/build_helper.py

```diff
@@ -156,31 +156,28 @@ def configured_build_targets(self) -> Set[str]:
         """
         Return the targets to build.
 
-        1. If the job is not defined, use the test_targets.
-        2. If the job is defined, but not the targets, use "fedora-stable" alias otherwise.
+        1. Use targets defined for build job and targets defined for test job.
+        2. Use "fedora-stable" alias if neither defined.
         """
-        if (
-            (not self.job_build or not self.job_build.metadata.targets)
-            and self.job_tests
-            and self.job_tests.metadata.targets
-        ):
-            return self.configured_tests_targets
+        targets = set()
+        if self.job_build:
+            targets.update(self.job_build.metadata.targets)
 
-        if self.job_build and self.job_build.metadata.targets:
-            return self.job_build.metadata.targets
+        if self.job_tests:
+            targets.update(self.job_tests.metadata.targets)
```

Comment:
I think this could be problem / ambiguous with the following configuration:

```yaml
jobs:
- job: copr_build
  trigger: pull_request

- job: tests
  trigger: pull_request
  metadata:
    targets:
    - fedora-rawhide-x86_64
```

Based on the previous description I'd expect COPR build for `fedora-stable` and tests for `fedora-rawhide`. (Based on code we built `fedora-rawhide` and ran tests for it.)
Based on the current description it should be just `fedora-rawhide`. (Code conforms to the description).

Is this intended change though?

I can imagine use case:
> I want COPR builds for `stable` and run tests for `rawhide`, so I would just *abuse* defaults for `copr_build` and expect the service to build the `rawhide` for TF too.

Reply:
Looking at the documentation my question might be answered. We "require" targets to be set :D

Reply:
But this is not really a change. 
Previous code from `configured_build_targets`:
```python
        if (
            (not self.job_build or not self.job_build.metadata.targets)
            and self.job_tests
            and self.job_tests.metadata.targets
        ):
            return self.configured_tests_targets
 ```
 
 On the other hand, it is true it can be misleading and we can change it now, if we decide or document it properly.